types.d.ts

```diff
@@ -0,0 +1,699 @@
+   * @param {String} actionId ID of the action
+   * @returns {Object} The requested action if found, else null
+   */
+  getAction(actionName: string, actionId: string): Action | null;
+
+  /**
+   * Add a new event and notify subscribers.
+   *
+   * @param {Object} event The event that occurred
+   */
+  addEvent(event: Event): void
+
+  /**
+   * Add an available event.
+   *
+   * @param {String} name Name of the event
+   * @param {Object} metadata Event metadata, i.e. type, description, etc., as
+   *                          an object.
+   */
+  addAvailableEvent(name: string, metadata?: Event.EventMetadata): void
+
+  /**
+   * Perform an action on the thing.
+   *
+   * @param {String} actionName Name of the action
+   * @param {Object} input Any action inputs
+   * @returns {Object} The action that was created.
+   */
+  performAction(actionName: string, input?: InputType): Action;
+
+  /**
+   * Remove an existing action.
+   *
+   * @param {String} actionName Name of the action
+   * @param {String} actionId ID of the action
+   * @returns boolean indicating the presence of the action.
+   */
+  removeAction(actionName: string, actionId: string): boolean;
+
+  /**
+   * Add an available action.
+   *
+   * @param {String} name Name of the action
+   * @param {Object} metadata Action metadata, i.e. type, description, etc., as
+   *                          an object.
+   * @param {Object} cls Class to instantiate for this action
+   */
+  addAvailableAction(name: string, metadata: Action.ActionMetadata, cls: InstanceType<ActionTypeClass>);
+
+  /**
+   * Add a new websocket subscriber.
+   *
+   * @param {Object} ws The websocket
+   */
+  addSubscriber(ws: Subscriber)
+
+  /**
+   * Remove a websocket subscriber.
+   *
+   * @param {Object} ws The websocket
+   */
+  removeSubscriber(ws: Subscriber)
+
+  /**
+   * Add a new websocket subscriber to an event.
+   *
+   * @param {String} name Name of the event
+   * @param {Object} ws The websocket
+   */
+  addEventSubscriber(name: string, ws: Subscriber)
+
+  /**
+   * Remove a websocket subscriber from an event.
+   *
+   * @param {String} name Name of the event
+   * @param {Object} ws The websocket
+   */
+  removeEventSubscriber(name: string, ws: Subscriber): void
+
+
+  /**
+   * Notify all subscribers of a property change.
+   *
+   * @param {Object} property The property that changed
+   */
+  propertyNotify(property: Property): void;
+
+  /**
+   * Notify all subscribers of an action status change.
+   *
+   * @param {Object} action The action whose status changed
+   */
+  actionNotify(action: Action): void;
+
+  /**
+   * Notify all subscribers of an event.
+   *
+   * @param {Object} event The event that occurred
+   */
+  eventNotify(event: Event): void;
+}
+
+declare namespace Thing {
+  interface ThingDescription {
+    name: string
+    href: string
+    '@context': string
+    '@type': string[]
+    properties: Property.PropertyDescription
```

Comment:
This would actually be `{ [name: string]: Property.PropertyDescription }` since it's the set of all property descriptions

Reply:
Corrected in https://github.com/mozilla-iot/webthing-node/pull/91/commits/8953a3de69b7bc643321bfc3b18f70d56d2f696c